### PR TITLE
feat(swap): show app fee details inline

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This shows the app fee details directly as a line item.

### Test plan

- Updated tests

| No quote | Valora fee disabled or 0 fee from API | Valora fee > 0 from API | Valora fee > 0 from API, but no USD price |
|----------|----------|----------|----------|
| image | image | image | image |
| image | image | image | image |

Notes: 
- copies are not final

### Related issues

- Fixes [RET-1061](https://linear.app/valora/issue/RET-1061)

### Backwards compatibility

Yes

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes
in this PR will:

- [x] Continue to work without code changes, OR trigger a compilation
error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/swap/SwapTransactionDetails.tsx [hunk 1]
```diff
@@ -194,15 +200,52 @@ export function SwapTransactionDetails({
...
+            })
+            appFeeInfoBottomSheetRef.current?.snapToIndex(0)
+          }}
+          label={t('swapScreen.transactionDetails.appFee')}
+          testID="SwapTransactionDetails/AppFee/MoreInfo"
+        />
+        <Text testID={'SwapTransactionDetails/AppFee'} style={styles.value}>
+          <Trans
+            i18nKey={'swapScreen.transactionDetails.appFeeValue'}
+            context={
```
Comment (bakoushin) [inline src/swap/SwapTransactionDetails.tsx:217]: this kind of feels too smart use for [context](https://www.i18next.com/translation-function/context) 😅

it implies this `context` prop conditions and underlying children rendering conditions must be kept in sync (e.g. both have to rely on `appFee.token.priceUsd`). i expect it to be hard/error-prone to quickly adjust in the future: one has to account for this, and the `translation.json` template as well.

wouldn't the good old conditional rendering without translation templates/context be less complex?

**Review by bakoushin — APPROVED** [review]
LGTM!

Left one nit about perceived complexity, more to have a discussion.

File: src/swap/SwapTransactionDetails.tsx (hunk 1 above)
Comment (jeanregisser) [reply src/swap/SwapTransactionDetails.tsx:217]: Yes it is a bit complex. But we need the ability to properly translate this.

The children conditionals aren't entirely needed, I did that to make it easier to understand tests.

We could have 4 very similar renders. Probably easier to understand but with more duplication. Though there's also test coverage.

In the interest of time, I can take a stab at this later to not block the current release further than needed. What do you think?

Comment (bakoushin) [reply src/swap/SwapTransactionDetails.tsx:217]: yes, I agree!
I'm not sure if we actually have to tune it given there is a refactoring of swap details is ahead

**Pull request merged**